src/parser/HLSTree.cpp

```diff
@@ -527,7 +527,7 @@ HLSTree::PREPARE_RESULT HLSTree::prepareRepresentation(Period* period,
         }
         else if (line.compare(0, 22, "#EXT-X-MEDIA-SEQUENCE:") == 0)
         {
-          newStartNumber = atol(line.c_str() + 22);
+          newStartNumber = atoll(line.c_str() + 22);
```

Comment:
```suggestion
          newStartNumber = static_cast<uint64_t>(std::atoll(line.c_str() + 22));
```
atoll is not unsigned

Reply:
I didn't add the cast as negative numbers in the HLS playlist would be illegal but you're right, better to show completeness here and remove any uncertainty from others reading.

Reply:
is not a problem of what value you parse but the type that it is different
unsigned type has a different range from signed, therefore this is not a good code
you assume that there will be no negative values
but if for any reason a negative value occurs, it will result in an unexpected positive value

Reply:
new change is good one!